Data/models/Domain.swift

```diff
@@ -32,27 +32,24 @@ public final class Domain: NSManagedObject, CRUD {
         super.awakeFromInsert()
     }
 
-    // Always use this function to save or lookup domains in the table
-    class func domainAndScheme(fromUrl url: URL?) -> String {
-        let domainUrl = (url?.scheme ?? "http") + "://" + (url?.normalizedHost ?? "")
-        return domainUrl
-    }
-
-    public class func getOrCreateForUrl(_ url: URL, context: NSManagedObjectContext) -> Domain? {
-        let domainString = Domain.domainAndScheme(fromUrl: url)
+    public class func getOrCreateForUrl(_ url: URL, context: NSManagedObjectContext) -> Domain {
+        let domainString = url.domainURL.absoluteString
         let fetchRequest = NSFetchRequest<NSFetchRequestResult>()
         fetchRequest.entity = Domain.entity(context)
         fetchRequest.predicate = NSPredicate(format: "url == %@", domainString)
-        var result: Domain?
+        var result: Domain!
         context.performAndWait {
             do {
-                let results = try context.fetch(fetchRequest) as? [Domain]
-                if let item = results?.first {
-                    result = item
+                var domain: Domain
+                let domains = try context.fetch(fetchRequest) as? [Domain]
```

Comment:
Replace fetch request and getting first domain with `CRUDProtocols.first(where: NSPredicate(format: "url == %@", domainString), context: context)`